src/guides/components/endpoint.md

```diff
@@ -0,0 +1,253 @@
+
+Add the following to your `/packages/frontend/src/index.ts` file:
+
+Again, we need to import the necessary type aliases: the base `SDK` and the `API` we just defined in the backend.
+
+``` ts
+import type { Caido } from "@caido/sdk-frontend";
+import type { API } from "../../backend/src/index.ts";
+```
+
+Next, we add the API to the SDK by creating the `CaidoSDK` type alias.
+
+``` ts
+export type CaidoSDK = Caido<API>;
+```
+
+The page will include two input fields, one for each expected number parameter and a `<p></p>` tag that will be used to display the result.
+
+::: tip
+For additional documentation on creating a page - click [here](/guides/components/page.md).
+:::
+
+``` ts
+const createPage = (sdk: CaidoSDK) => {
+
+    const resultText = document.createElement("p");
+    resultText.textContent = "Result will appear here.";
+
+    const inputA = document.createElement("input");
+    inputA.type = "number";
+    inputA.value = "0";
+    inputA.style.color = "black";
+    
+    const inputB = document.createElement("input");
+    inputB.type = "number";
+    inputB.value = "0";
+    inputB.style.color = "black";
+```
+
+The page also includes a button, that when clicked will take the supplied input and use the values as the parameters for the `multiply` function which is called using `sdk.backend.multiply(a, b);`. To account for processing time, `await` is used.
+
+``` ts
+    const calculateButton = sdk.ui.button({
+        variant: "primary",
+        label: "Calculate",
+    });
+
+    calculateButton.addEventListener("click", async () => {
+        const a = Number(inputA.value);
+        const b = Number(inputB.value);
+        const result = await sdk.backend.multiply(a, b);
+        resultText.textContent = `Result: ${result}`;
+    });
+```
+
+The page elements are all grouped together and stored in the `container` variable which is then used as the `body` property value of the `sdk.ui.card({})` method. The page is then added to Caido using the `sdk.navigation.addPage()` method with the path of `"/multiply-page"`. Its content is the `card` we just defined.
+
+``` ts
+    const container = document.createElement("div");
+    container.appendChild(inputA);
+    container.appendChild(inputB);
+    container.appendChild(calculateButton);
+    container.appendChild(resultText);
+
+    const card = sdk.ui.card({
+        body: container
+    });
+
+    sdk.navigation.addPage("/multiply-page", {
+        body: card
+    });
+}
+```
+
+Finally, we define an initialization function for the frontend that will generate this page when Caido loads our plugin. The `sdk.sidebar.registerItem()` method will add a tab for our plugin page to the left-hand side menu of Caido named `Multiply` with a calculator icon.
+
+``` ts
+export function init(sdk: CaidoSDK) {
+    createPage(sdk);
+    
+    sdk.sidebar.registerItem("Multiply", "/multiply-page", {
+        icon: "fas fa-calculator"
+    });
+}
+```
+
+::: tip
```

Comment:
Same comment regarding code snippets that need to be expanded. Let's bring that to the top of the section.